File: js/emit_exponent.js
function
emit_exponent(u, p)
{
	var v;

	if (isnum(p) && !isnegativenumber(p)) {
		emit_numeric_exponent(u, p); // sign is not emitted
		return;
	}

	v = {type:SUPERSCRIPT, a:[], level:u.level + 1};

	emit_expr(v, p);

	emit_update_superscript(u, v);

	u.a.push(v);
}

File: app/emit_exponent.c
#include "app.h"

void
emit_exponent(struct atom *p)
{
	if (isnum(p) && !isnegativenumber(p)) {
		emit_numeric_exponent(p); // sign is not emitted
		return;
	}

	emit_level++;
	emit_list(p);
	emit_level--;

	emit_update_superscript();
}
